Metrics.py

```diff
@@ -0,0 +1,88 @@
+import os
+from github import Github
+from datetime import datetime, timedelta, timezone
+from collections import defaultdict
+import json
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
+
+# Constants
+ORG_NAME = "anuket-project"
+TOKEN = os.getenv('GITHUB_TOKEN')
+TIMEFRAME_DAYS = 365  # Change this to filter the timeframe, e.g., 60 for last 2 months
+AFFILIATION_FILE = "affiliations.json"
+
+# Load affiliations
+with open(AFFILIATION_FILE, 'r') as file:
+    affiliations = json.load(file)
+
+# Initialize Github object
+g = Github(TOKEN)
+org = g.get_organization(ORG_NAME)
+repos = org.get_repos()
+
+# Calculate timeframe
+end_date = datetime.now(timezone.utc)
+start_date = end_date - timedelta(days=TIMEFRAME_DAYS)
+
+def get_repo_metrics(repo):
+    commits = repo.get_commits(since=start_date, until=end_date)
+    commit_data = []
+
+    for commit in commits:
+        commit_author = commit.author.login if commit.author else 'unknown'
+        commit_data.append({
+            'sha': commit.sha,
+            'author': commit_author,
+            'date': commit.commit.author.date,
+        })
+
+    # Number of commits
+    num_commits = len(commit_data)
+
+    # Number of committers
+    committers = set([commit['author'] for commit in commit_data])
+    num_committers = len(committers)
+
+    # Number of committer companies
+    committer_companies = set([affiliations.get(committer, 'unknown') for committer in committers])
+    num_committer_companies = len(committer_companies)
+
+    # Top 50% committers by commits
+    commit_count_by_committer = defaultdict(int)
+    for commit in commit_data:
+        commit_count_by_committer[commit['author']] += 1
+
+    sorted_committers = sorted(commit_count_by_committer.items(), key=lambda item: item[1], reverse=True)
+    top_50_percent_committers = sorted_committers[:len(sorted_committers) // 2]
+    num_top_50_percent_committers = len(top_50_percent_committers)
+
+    # Number of committer companies for top 50% committers
+    top_50_percent_committer_companies = set([affiliations.get(committer[0], 'unknown') for committer in top_50_percent_committers])
+    num_top_50_percent_committer_companies = len(top_50_percent_committer_companies)
+
+    # Output the metrics
+    metrics = {
+        "num_commits": num_commits,
+        "num_committers": num_committers,
+        "num_committer_companies": num_committer_companies,
+        "num_top_50_percent_committers": num_top_50_percent_committers,
+        "num_top_50_percent_committer_companies": num_top_50_percent_committer_companies,
+    }
+    
+    return metrics
+
+# Collect metrics for all repos
+all_repo_metrics = {}
```

Comment:
Create an entry point as it is described in the first example [here](https://amir.rachum.com/python-entry-points/).

When that is done you can move the function definitions behind the main logic of the code.

Reply:
Updated Please Check it